ultralytics/data/explorer/explorer.py

```diff
@@ -0,0 +1,158 @@
+            h0, w0 = im.shape[:2]  # orig hw
+            return im, (h0, w0), im.shape[:2]
+
+        return self.ims[i], self.im_hw0[i], self.im_hw[i]
+
+    def build_transforms(self, hyp=None):
+        transforms = Format(
+            bbox_format='xyxy',
+            normalize=False,
+            return_mask=self.use_segments,
+            return_keypoint=self.use_keypoints,
+            batch_idx=True,
+            mask_ratio=hyp.mask_ratio,
+            mask_overlap=hyp.overlap_mask,
+        )
+        return transforms
+
+
+class Explorer:
+
+    def __init__(self, data='coco128.yaml', model='yolov8n.pt', uri='~/ultralytics/explorer') -> None:
+        self.connection = lancedb.connect(uri)
+        self.table_name = Path(data).name
+        self.model = YOLO(model)
+        self.data = data  # None
+        self.choice_set = None
+
+        self.table = None
+
+    def create_embeddings_table(self, force=False, split='train', verbose=False):
+        if (self.table is not None and not force):
+            logger.info('Table already exists. Reusing it. Pass force=True to overwrite it.')
+            return
+        if self.table_name in self.connection.table_names():
+            logger.info(f'Table {self.table_name} already exists. Reusing it. Pass force=True to overwrite it.')
+            self.table = self.connection.open_table(self.table_name)
+            return
+        if self.data is None:
+            raise ValueError('Data must be provided to create embeddings table')
+
+        data_info = check_det_dataset(self.data)
+        if split not in data_info:
+            raise ValueError(
+                f'Split {split} is not found in the dataset. Available keys in the dataset are {list(data_info.keys())}'
+            )
+
+        choice_set = data_info[split]
+        choice_set = choice_set if isinstance(choice_set, list) else [choice_set]
+        self.choice_set = choice_set
+
+        dataset = ExplorerDataset(img_path=choice_set, data=data_info, augment=False, cache=False)
+
+        # Create the table schema
+        schema = pa.schema([
+            pa.field('im_file', pa.string()),
+            pa.field('labels', pa.list_(pa.string())),
+            pa.field('bboxes', pa.list_(pa.list_(pa.float32()))),
+            pa.field('cls', pa.list_(pa.int32())),
+            pa.field('vector', pa.list_(pa.float32(),
+                                        self.model.embed(dataset[0]['im_file'])[0].shape[0])), ])
+        table = self.connection.create_table(self.table_name, schema=schema, mode='overwrite')
+        table.add(
+            self._yeild_batches(dataset,
+                                data_info,
+                                self.model,
+                                exclude_keys=['img', 'ratio_pad', 'resized_shape', 'ori_shape', 'batch_idx']))
+
+        self.table = table
+
+    @staticmethod
+    def _yeild_batches(dataset, data_info, model, exclude_keys: List):
```

Comment:
Is there a typo? Did you mean `_yield_batches`?

Reply:
Yes. This is still is WIP